File: src/main/java/org/jukeboxmc/block/BlockSweetBerryBush.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemSweetBerryBush;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockSweetBerryBush extends Block {

    public BlockSweetBerryBush() {
        super( "minecraft:sweet_berry_bush" );
    }

    @Override
    public ItemSweetBerryBush toItem() {
        return new ItemSweetBerryBush();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.SWEET_BERRY_BUSH;
    }

    public void setGrowth( int value ) { //0-7
        this.setState( "growth", value );
    }

    public int getGrowth() {
        return this.stateExists( "growth" ) ? this.getIntState( "growth" ) : 0;
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockSweetBerryBush.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.SweetBerryBush
import org.jukeboxmc.api.item.Item
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.api.item.enchantment.EnchantmentType
import org.jukeboxmc.server.block.JukeboxBlock
import kotlin.random.Random

class BlockSweetBerryBush(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    SweetBerryBush {

    override fun canPassThrough(): Boolean {
        return true
    }

    override fun getGrowth(): Int {
        return this.getIntState("growth")
    }

    override fun setGrowth(value: Int): SweetBerryBush {
        return this.setState("growth", value)
    }

    override fun getDrops(item: Item): MutableList<Item> {
        var level = 0
        if (item.hasEnchantment(EnchantmentType.FORTUNE)) {
            level = item.getEnchantment(EnchantmentType.FORTUNE)!!.getLevel()
        }
        val growth = this.getGrowth()
        return if (growth < 3) {
            mutableListOf()
        } else if (growth in 3..6) {
            mutableListOf(Item.create(ItemType.SWEET_BERRIES, 1 + Random.nextInt(2) + level))
        } else {
            mutableListOf(Item.create(ItemType.SWEET_BERRIES, 2 + Random.nextInt(2) + level))
        }
    }
}